extensions/sockets/src/test/java/org/mule/extension/socket/TcpSendAndReceiveStringTestCase.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.socket;
+
+import static org.junit.Assert.assertEquals;
+import org.mule.module.socket.api.protocol.DirectProtocol;
+import org.mule.module.socket.api.protocol.EOFProtocol;
+import org.mule.module.socket.api.protocol.LengthProtocol;
+import org.mule.module.socket.api.protocol.SafeProtocol;
+import org.mule.module.socket.api.protocol.StreamingProtocol;
+import org.mule.module.socket.api.protocol.TcpProtocol;
+import org.mule.module.socket.api.protocol.XmlMessageEOFProtocol;
+import org.mule.module.socket.api.protocol.XmlMessageProtocol;
+import org.mule.runtime.core.util.IOUtils;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TcpSendAndReceiveStringTestCase extends SocketExtensionTestCase
+{
+
+    public static final String RESPONSE_TEST_STRING = TEST_STRING + "_modified";
+
+    @Rule
+    public DynamicPort port1 = new DynamicPort("port_tcp");
+
+    @Parameterized.Parameter(0)
+    public TcpProtocol tcpProtocol;
+
+    @Parameterized.Parameter(1)
+    public String testName;
+
+    @Parameterized.Parameters(name = "{1}")
+    public static Collection<Object[]> data()
+    {
+        return Arrays.asList(new Object[][] {
+                {new LengthProtocol(), LengthProtocol.class.getSimpleName()},
```

Comment:
Already seen this repeated list.. can't you have a base parameterized test class which defines it? Otherwise you have to change a lot of tests in the eventuality of a new protocol... At the very least you should be able to have the data array reusable